### PR TITLE
fix: coolline does not occupy all space when class has no stance bar

### PR DESCRIPTION
**Pull request by abdes**
When the stance bar is present, anchor CoolLine to the stance bar, otherwise, anchor it to the tukui action bar 1. In all cases anchor it to tukui action bar 3 on the right side.

### CODE REVIEW

File: SanUI/Modules/AddonMods/coolline.lua [hunk 1]
```diff
@@ -34,11 +34,11 @@ function S.modCoolLine(event)
 			CoolLine:ClearAllPoints()
 			if TukuiStanceBar and NumForms > 0 then 
 				CoolLine:SetPoint("BOTTOMLEFT",TukuiStanceBar,"BOTTOMRIGHT",spacing,spacing)
-				CoolLine:SetPoint("TOPLEFT",TukuiStanceBar,"TOPRIGHT",spacing,-spacing)
```
Comment (SanUI) [inline SanUI/Modules/AddonMods/coolline.lua]: Why are you removing this?

File: SanUI/Modules/AddonMods/coolline.lua [hunk 2]
```diff
@@ -34,11 +34,11 @@ function S.modCoolLine(event)
...
 			else
+				CoolLine:SetPoint("BOTTOMLEFT",TukuiActionBar1,"TOPLEFT",0,spacing)
 				CoolLine:SetPoint("TOPRIGHT",TukuiActionBar3,"TOPLEFT",-spacing,-spacing)
 			end
-			CoolLine:SetPoint("BOTTOMRIGHT",TukuiActionBar3,"BOTTOMLEFT",-spacing,spacing)
+			CoolLine:SetPoint("BOTTOMRIGHT",TukuiActionBar3,"BOTTOMLEFT",-spacing,-spacing)
```
Comment (SanUI) [inline SanUI/Modules/AddonMods/coolline.lua]: What's the reason for this change?

**Comment by abdes** [comment]
When a class has no stance bar, coolline does not occupy the space
available. My change is to anchor it on the left side to the bottom bar so
it extends to occupy the empty space.

**Comment by SanUI** [comment]
In understand the reason for this PR, but I don't understand those individual lines. The first one is in the if-arm for classes with a stance bar, why would that need a change? The second one makes a change for all classes, why is this? I'd have expected for you to only change the "else" arm.

File: SanUI/Modules/AddonMods/coolline.lua (hunk 1 above)
Comment (abdes) [reply SanUI/Modules/AddonMods/coolline.lua]: We don't need a TOPLEFT anchor if we have a TOPRIGHT anchor on action bar 3 just a few lines below...

https://github.com/SanUI/SanUI/blob/100ef0af3b2dcf038d2272250b1a87abcfd2ad9d/SanUI/Modules/AddonMods/coolline.lua#L39

File: SanUI/Modules/AddonMods/coolline.lua (hunk 2 above)
Comment (abdes) [reply SanUI/Modules/AddonMods/coolline.lua]: pixel perfect symmetrical alignment. Spacing above and below of coolline. See BOTTOMRIGHT anchor...

**Comment by abdes** [comment]
### Class with stance bar:

image

### Class with no stance bar:

image

File: SanUI/Modules/AddonMods/coolline.lua (hunk 1 above)
Comment (SanUI) [reply SanUI/Modules/AddonMods/coolline.lua]: Wait, am I missing something? That's a BOTTOMRIGHT you're referring to, isn't it? Line 41, right?

File: SanUI/Modules/AddonMods/coolline.lua (hunk 2 above)
Comment (SanUI) [reply SanUI/Modules/AddonMods/coolline.lua]: Oh I get it, but it was intentional this way, I wanted it aligned with the buttons themselves, not the bar backgrounds (which is what `TukuiActionBar3` is referring to). I'd prefer if you did not change this.

And if you look closely, in the screenshot without stance bar, it does look off in exactly this way.

Comment (abdes) [reply SanUI/Modules/AddonMods/coolline.lua]: If this is how you want it,
image

then the code should be:
```lua
CoolLine:ClearAllPoints()
if TukuiStanceBar and NumForms > 0 then 
	CoolLine:SetPoint("TOPLEFT",TukuiStanceBar,"TOPRIGHT",spacing,-spacing)
	CoolLine:SetPoint("BOTTOMRIGHT",TukuiActionBar3,"BOTTOMLEFT",-spacing,spacing)
else
	CoolLine:SetPoint("LEFT",TukuiActionBar1,"LEFT")
	CoolLine:SetPoint("BOTTOM",TukuiActionBar3,"BOTTOM", 0, spacing)
	CoolLine:SetPoint("TOPRIGHT",TukuiActionBar3,"TOPLEFT",-spacing,-spacing)
end
S.placeStanceBar()
```

**Comment by abdes** [comment]
### Class with stance bar
image

### Class with no stance bar
image

CoolLine aligned exactly on the buttons of Action Bar 1 and Stance Bar if any.

**Comment by SanUI** [comment]
Ok, that looks more like what I'd suspected, nice. Although I'm a bit surprised the code for non-stancebar works...

I have to apologize, but I need to push an update to master which will conflict with this PR. If you don't care to rebase, I can just add a commit myself with your code. Sorry for the hassle ;)

**Comment by abdes** [comment]
> Ok, that looks more like what I'd suspected, nice. Although I'm a bit surprised the code for non-stancebar works...
> 
> I have to apologize, but I need to push an update to master which will conflict with this PR. If you don't care to rebase, I can just add a commit myself with your code. Sorry for the hassle ;)

No problem.

The code actually works because of the way the anchoring works in WoW. You can do it in multiple ways by setting constraints. It does not work all the time, but this one I tested it :-)
